README.md

```diff
@@ -40,13 +40,21 @@ While the middleware does attempt to leave request-promise as pristine as possib
 
 ## Examples
 
-Once you've defined your middleware, you can simply register it by creating a new framework object, and then getting the request object:
+Once you've defined your middleware, you can simply register it by creating a new framework object, and then getting the request object. If you want to use native Promises, you can do so as follows:
 ```
-var rpMiddlewareFramework = new RequestPromiseMiddlewareFramework(require("request-promise"), middleware);
-var rp = rpMiddlewareFramework.getMiddlewareEnabledRequestPromise();
+const rpMiddlewareFramework = new RequestPromiseMiddlewareFramework({ rp: require("request-promise") }, middleware);
+const rp = rpMiddlewareFramework.getMiddlewareEnabledRequestPromise();
 ```
 
-You can then use returned ``rp`` object just like you normally would.
+Or, if you want to use an alternate Promise library such as bluebird, you can do so as follows:
+
+```
+const bluebirdPromise = require("bluebird");
+const rpMiddlewareFramework = new RequestPromiseMiddlewareFramework({ rp: require("request-promise"), P: bluebirdPromise }, middleware);
```

Comment:
`P` is not a great name for this override. I'll look at some common libraries and how they allow for it:

* [request-promise-any](https://github.com/request/request-promise-any) (maybe we should just use this. It is endorsed by [request-promise](https://github.com/request/request-promise)), which uses [any-promise](https://github.com/kevinbeaty/any-promise): field is named `Promise`
* [AWS SDK](https://aws.amazon.com/blogs/developer/support-for-promises-in-the-sdk/): `setPromiseDependency` (As an option, I'd infer it to `promiseDependency`)
* [mongoose](https://mongoosejs.com/docs/connections.html): `promiseLibrary`

Reply:
The `any-promise` stuff is intriguing although it replaces one required dependency (bluebird) with another (any-promise), and it requires users of the library to know to register a Promise dependency with any-promise (which they might not do). With the implementation I used at least it's deterministic which Promise library is chosen. That said, I switched from P to PromiseDependency.